### PR TITLE
Fixed repo creation bug

### PR DESCRIPTION
**Pull request by JA-04**
- Fixed the repo creation bug which occurred due to different path structure of different operating systems.
- Added error handling for repo creation inside organizations using SAML SSO.
- Added telemetry error logging for simple-git functions.
- Changed visibility of newly created repos to private.

### CODE REVIEW

File: src/configure/clients/restClient.ts [hunk 1]
```diff
@@ -20,6 +20,21 @@ export class RestClient extends ServiceClient {
         });
     }
 
+    public sendRequest3<TResult>(options: PathTemplateBasedRequestPrepareOptions | UrlBasedRequestPrepareOptions): Promise<TResult> {
```
Comment (bishal-pdMSFT) [inline src/configure/clients/restClient.ts:23]: why do we need a new sendrequest implementation?

File: src/configure/clients/github/githubClient.ts [hunk 2]
```diff
@@ -36,41 +36,38 @@ export class GithubClient {
...
+                "has_wiki": true
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+            .then((detail: GitHubRepo) => {
+                return detail;
+            }).catch(error => {
+                telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
+                if (error.response.statusCode === 422) {
```
Comment (bishal-pdMSFT) [inline src/configure/clients/github/githubClient.ts:66]: when does this happen?

File: src/configure/clients/github/githubClient.ts [hunk 3]
```diff
@@ -36,41 +36,38 @@ export class GithubClient {
...
                     return null;
-                });
-        }
-        catch (error) {
-            telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
-            return null;
-        }
+                }
```
Comment (bishal-pdMSFT) [inline src/configure/clients/github/githubClient.ts:68]: should you throw an error rather than returning null. This would ensure consistency in the function's behavior in case something goes wrong. Right now it is throowing error in some case and returning null in some.

File: src/configure/clients/restClient.ts (hunk 1 above)
Comment (JA-04) [reply src/configure/clients/restClient.ts:23]: The API call for creating new repo can fail due to multiple reasons, including the existence of repo of the same name. In case of an existing repo name, we need to make API call again with "local repo name + org name" but in case of any other failure, the process needs to be terminated. For deciding the right condition for termination, error code is needed that is why the entire `response` needs to be returned instead of `response.body`.

File: src/configure/clients/github/githubClient.ts (hunk 2 above)
Comment (JA-04) [reply src/configure/clients/github/githubClient.ts:66]: It happens when repo creation fails due to the existence of another repo of same name.

File: src/configure/clients/github/githubClient.ts (hunk 3 above)
Comment (JA-04) [reply src/configure/clients/github/githubClient.ts:68]: `null` is returned only in the case where API call needs to be made again (same repo name exists). If the error is directly thrown then in the calling function there will be no error code to check the reason for failure.

File: src/configure/clients/github/githubClient.ts [hunk 4]
```diff
@@ -36,41 +36,39 @@ export class GithubClient {
...
+                "description": "Repo created from VScode extension 'Deploy to Azure'",
+                "homepage": "https://github.com",
+                "private": true,
+                "has_issues": true,
+                "has_projects": true,
+                "has_wiki": true
+            },
+            deserializationMapper: null,
+            serializationMapper: null,
+            sendCompletetResponse: true
```
Comment (bishal-pdMSFT) [inline src/configure/clients/github/githubClient.ts:61]: rename to `returnFullResponseForFailure`

File: src/configure/helper/commonHelper.ts [hunk 5]
```diff
@@ -93,18 +93,18 @@ export async function executeFunctionWithRetry(
...
-            try {
-                let result = await func();
-                return result;
-            }
-            catch (error) {
-                internalError = error;
-                logger.log(JSON.stringify(error));
-                await Q.delay((resolve) => {resolve();}, retryIntervalTimeInSec * 1000);
-            }
+    let internalError = null;
```
Comment (bishal-pdMSFT) [inline src/configure/helper/commonHelper.ts:96]: revert this change

File: src/configure/clients/github/githubClient.ts (hunk 4 above)
Comment (JA-04) [reply src/configure/clients/github/githubClient.ts:61]: Done

File: src/configure/helper/commonHelper.ts (hunk 5 above)
Comment (JA-04) [reply src/configure/helper/commonHelper.ts:96]: Okay
